### PR TITLE
Make cve module self contained and not require stdlib

### PR DESCRIPTION
**Pull request by jeffmccune**
Without this patch, the cve20113872 module made use of the validate_re()
function from stdlib.  To minimize the number of steps the end user
needs to make this patch copies the function into this module.

This means stdlib is no longer required and the function has been
renamed to prevent conflicts with stdlib should it already be installed
which is the case on PE 1.2.3.

In addition, the README, module metadata and Rakefile have been updated
to reflect this change.  The rake tasks have also been modified to
function if rspec is not installed, which is the case on a PE system.

Finally, a build and clean task have been added to help clean out and
rebuild the cve module packages.

### CODE REVIEW

**Comment by jhelwig** [comment]
Looking at this one.

File: Modulefile [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 version '0.0.1'
-source 'git@git.puppetlabs.net:cve20113872.git'
+source 'git@github.com:puppetlabs/puppetlabs-cve20113872.git'
 author 'puppetlabs'
 license 'Apache 2.0'
-summary 'Fix and Remedy CVE-2011-3872'
+summary 'Migration Assistant for CVE-2011-3872'
 description 'This module provides tools to help close the security risk documented in CVE-2011-3872 and migrate a Puppet infrastructure to a new Certificate Authority.'
-project_page 'http://www.puppetlabs.com'
+project_page 'https://github.com/puppetlabs/puppetlabs-cve20113872'
```
Comment (jhelwig) [inline Modulefile:8]: Seems like everything from here up should be in a separate commit?

**Comment by jhelwig** [comment]
I'm a bit confused by the changes to `bin/pe_step1_secure_the_master` since they don't appear to be discussed anywhere in the commit message, also I'm not sure how the README changes are related to the stdlib change.

File: Modulefile (hunk 1 above)
Comment (jeffmccune) [inline Modulefile:8]: Agreed, I'll pull it out into its own.

**Comment by jeffmccune** [comment]
Yeah, I modified the step1 script because it was stoping the service before it checked if there was a backup already.  This meant the service would stay down if we bail out.

I'll ammend the commit message.

**Comment by jeffmccune** [comment]
@jhelwig I cleaned up this topic branch.  Please re-review and merge if it's good now.

**Pull request merged**